pkg/querier/queryrange/split_by_interval.go

```diff
@@ -170,6 +169,17 @@ func (h *splitByInterval) Do(ctx context.Context, r queryrange.Request) (queryra
 		return h.next.Do(ctx, r)
 	}
 
+	// Align the queries time range to multiples of the interval to allow for better caching.
+	ims := interval.Milliseconds()
```

Comment:
Can we create a function that takes a Request and returns a list of request ? This way it'll make tests easier to read without having to deal with middleware ?

Reply:
I guess it should be used inside the splitter... or replace the splitter.

Reply:
I moved the alignment code to it's own function but kept it independent of the `splitter` since it looks like we have to different splitter functions and I'm not sure if we only wanted the alignment code in one, or if there might be more splitter functions in the future.